include/macro.inc

```diff
@@ -1,4 +1,65 @@
+# glabel
+
 .macro glabel label
     .global \label
     \label:
 .endm
+
+.macro BEGIN label
+    glabel \label
+    .ent \label
+.endm
+
+.macro END label
+    .type \label, @function
+    .size \label, . - \label
+    .end \label
+.endm
+
+.macro BEGINDATA label
+    glabel \label
+.endm
+
+.macro ENDDATA label
+    .type \label, @object
+    .size \label, . - \label
+.endm
+
+# cop0 register aliases
+
+.set Index,         $0
+.set Random,        $1
+.set EntryLo0,      $2
+.set EntryLo1,      $3
+.set Context,       $4
+.set PageMask,      $5
```

Comment:
Could you explain the context of these variables that are set? Are these generic names or unique ones to `code_800D71F0.s`

Reply:
Those are the register names of the coprocessor 0 (a.k.a. Cop0).

From what I understand, this file does not need them, but I think Elliptic just took the whole file from Tharo's PR, probably to avoid conflicts.

Reply:
Yeah, I needed the macros so I took the whole file to mitigate conflicts as far as possible.